apps/dashboard/features/holders-and-delegates/hooks/useDelegates.ts

```diff
@@ -373,14 +380,21 @@ export const useDelegates = ({
     pagination.startCursor,
     orderBy,
     orderDirection,
+    address,
     isPaginationLoading,
   ]);
 
   // Enhanced refetch that resets pagination
   const handleRefetch = useCallback(() => {
     setCurrentPage(1);
-    refetch();
-  }, [refetch]);
+    refetch({
+      after: undefined,
+      before: undefined,
+      orderBy,
+      orderDirection,
+      ...(address && { addresses: [address] }),
+    });
+  }, [refetch, orderBy, orderDirection, address]);
 
   return {
```

Comment:
Missing error handling for the address filter parameter. If an invalid address format is passed, it could cause GraphQL query errors. Add validation for the address parameter format before including it in the query.